### ISSUE TITLE
bug with 'bite' and '@' in mininotation

### ISSUE THREAD

**Issue by TylerMclaughlin** [tidalcycles/Tidal#1053]
I'm on tidal_version 1.9.4.  I'm having a bug when I use `bite` with `@` symbol in mininotation.

Here's shorthand I use for midi:

```haskell
let m = sound "midi" # midichan 0
```

Take the following melody:
```haskell
d1 $ n "0 4 7" # m
```
For a negative control: I parameterize bite so it does nothing. The above code sounds exactly like
```haskell
d1 $ bite 3 "0 1 2" $ n "0 4 7" # m
```

Good, this test passed.

(it's dividing the pattern into 3 and patterning those chunks in the normal order).

Modifying the beat in this kind of way works in `bite`:

```haskell
d1 $ bite 3 "[0!5] [1!3] [2!9]" $ n "0 4 7" # m
```

But then this fails:

```haskell
d1 $ bite 3 "0@5 1@6 2@5" $ n "0 4 7" # m
```
It adds extra onsets for some reason, instead of just shifting the three notes around.  It doesn't sound like

```haskell
d1 $ n "0@5 4@6 7@5" # m
```

as I'd expect it to.

**Comment by TylerMclaughlin** [comment]
Yes.  I can reproduce your output examples, so things look good in the Haskell side.  
Though MIDI monitor and external devices are definitely showing extra notes.  Patterns that are identical after function application are producing different MIDI messages.  Very strange.

**Comment by yaxu** [comment]
Are you able to create the simplest example that creates extra triggers, and share midi output for that? Are you monitoring midi straight out of superdirt?

**Comment by TylerMclaughlin** [comment]
Smart to check later arcs @jwaldmann !  

A closer look on the output-- it looks deterministic (thank heavens), but complex.  Definitely not doing what I thought it would be doing with '@'.  

@yaxu supercollider is sending MIDI through my IAC driver so @jwaldmann is reproducing this with a different virtual MIDI device.  

Here's a minimal example that produces the behavior:

```haskell
bite 2 "0@2 1" $ n "0 7"
```

  I'd expect this to make `n "0@2 7"` but it makes `n "[0@2 7 0 0 7]/2"` instead.

**Comment by yaxu** [comment]
Yes this looks like a bug!

**Comment by yaxu** [comment]
Over a year later, this is still a problem:

```hs
drawLine $ bite 2 "0@2 1" $ "a b"
-- |a-b|aab|a-b|aab|a-b|aab|a-b|aab|a-b|aab|a-b|aab|a-b|aab|a-b|aab|a-b|aab|a-b
```

Indeed @jwaldmann the order of events in the query result shouldn't matter.

**Comment by yaxu** [comment]
Interestingly, strudel agrees:
```js
"<a b c d e> <1 2 3 4 5>".bite(2, "0 1@2").drawLine()
// |a1-|d23|b4-|e51|c2-|a34|d5-|b12|e3-|c45|a1-|d23|b4-|e51|c2-
```

`squeezeJoin` seems to be the culprit:
```hs
drawLine $ squeezeJoin $ struct "t t@2" $ pure ("<a b c d e> <1 2 3 4 5>" :: Pattern Char)
-- |a1a-1-|d42-c-|b2d-4-|e55-a-|c3b-2-|a13-d-|d4e-5-|b21-b-|e5c-3-|c34-e-|a1a-1-
```

**Comment by yaxu** [comment]
`focusArc`, used by `squeezeJoin` seems wrong:
```hs
let a = Arc (4%3) 2
  in queryArc (focusArc a ("a b c d" :: Pattern Char)) a
-- [(1⅔>1⅚)|'a',(1⅚>2)|'b',(1⅓>1½)|'c',(1½>1⅔)|'d']
```

**Comment by yaxu** [comment]
Yep that was the problem.. A pretty serious bug, surprised it hasn't let itself known more often! Thanks for reporting @TylerMclaughlin and for the nudge @sss-create
